src/services/payment-flow/index.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  CouldNotFindLightningPaymentFlowError,
+  CouldNotUpdateLightningPaymentFlowError,
+  NoExpiredLightningPaymentFlowsError,
+  UnknownRepositoryError,
+} from "@domain/errors"
+import { PaymentFlow } from "@domain/payments"
+import { WalletCurrency } from "@domain/shared"
+
+import { PaymentFlowState } from "./schema"
+
+export const PaymentFlowStateRepository = (
+  expiryTimeInSeconds: Seconds,
+): IPaymentFlowRepository => {
+  const persistNew = async <S extends WalletCurrency, R extends WalletCurrency>(
+    paymentFlow: PaymentFlow<S, R>,
+  ): Promise<PaymentFlow<S, R> | RepositoryError> => {
+    try {
+      const rawPaymentFlowState = rawFromPaymentFlow(paymentFlow)
+      const paymentFlowState = new PaymentFlowState(rawPaymentFlowState)
+      await paymentFlowState.save()
+      return paymentFlowFromRaw(paymentFlowState)
+    } catch (err) {
+      return new UnknownRepositoryError(err)
+    }
+  }
+
+  const findLightningPaymentFlow = async <S extends WalletCurrency>({
+    walletId,
+    paymentHash,
+    inputAmount,
+  }: {
+    walletId: WalletId
+    paymentHash: PaymentHash
+    inputAmount: BigInt
+  }): Promise<PaymentFlow<S, WalletCurrency> | RepositoryError> => {
+    try {
+      const result = await PaymentFlowState.findOne({
+        senderWalletId: walletId,
+        paymentHash,
+        inputAmount: Number(inputAmount),
+      })
+      if (!result) return new CouldNotFindLightningPaymentFlowError()
+
+      const paymentFlow: PaymentFlow<S, WalletCurrency> = paymentFlowFromRaw(result)
+      if (isExpired({ paymentFlow, expiryTimeInSeconds })) {
+        await deleteLightningPaymentFlow({ walletId, paymentHash, inputAmount })
```

Comment:
No need for await here or? Could be fire + forget.

Reply:
No need I think: https://github.com/GaloyMoney/galoy/pull/1237/commits/3bfaa39de68a7c9dbdea197d8044b53fab077817